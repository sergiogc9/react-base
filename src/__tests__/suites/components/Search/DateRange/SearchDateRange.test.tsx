import { shallow } from "enzyme";
import moment from 'moment';
import React from "react";
import SearchDateRange from "@src/components/Search/DateRange/SearchDateRange";
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const user = TestHelper.getUser();

let onSetBeginDate: any;
let onSetEndDate: any;

const beginDate = moment().startOf('day').subtract(1, 'week').toDate();
const endDate = moment().endOf('day').toDate();

describe("<SerachDateRange />", () => {
	let wrapper: any;

	beforeEach(() => {
		onSetBeginDate = jest.fn();
		onSetEndDate = jest.fn();
		wrapper = shallow(<SearchDateRange
			user={user}
			beginDate={null}
			endDate={null}
			onSetBeginDate={onSetBeginDate}
			onSetEndDate={onSetEndDate}
		/>);
	});

	it("Test component snapshot", () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Test with custom date ranges", () => {
		wrapper.setProps({ beginDate, endDate });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Test date pickers change ", () => {
		wrapper.setProps({ searchPeriod: "custom", beginDate, endDate });
		expect(onSetBeginDate).toHaveBeenCalledTimes(0);
		wrapper.find('#searchDateRangeBeginDate').at(0).props().onChange(new Date());
		expect(onSetBeginDate).toHaveBeenCalledTimes(1);

		expect(onSetEndDate).toHaveBeenCalledTimes(0);
		wrapper.find('#searchDateRangeEndDate').at(0).props().onChange(new Date());
		expect(onSetEndDate).toHaveBeenCalledTimes(1);
	});
});
