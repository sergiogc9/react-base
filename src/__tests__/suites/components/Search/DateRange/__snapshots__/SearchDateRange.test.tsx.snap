// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SerachDateRange /> Test component snapshot 1`] = `""`;

exports[`<SerachDateRange /> Test with custom date ranges 1`] = `
"<div id=\\"searchDateRange\\">
  <DatePickerContainer id=\\"searchDateRangeBeginDate\\" label=\\"filters.period.custom.begin_date\\" okLabel=\\"filters.period.custom.dialog.ok\\" cancelLabel=\\"filters.period.custom.dialog.cancel\\" fullWidth={false} portal={true} icon={{...}} defaultValue={{...}} value={{...}} locales=\\"en-GB\\" minDate={{...}} maxDate={{...}} onChange={[Function: onChange]} animateInline={true} previousIcon={{...}} nextIcon={{...}} autoOk={false} yearsDisplayed={100} defaultCalendarMode=\\"calendar\\" DateTimeFormat={[Function: DateTimeFormat]} okPrimary={true} cancelPrimary={true} closeOnEsc={true} closeYearOnSelect={false} disableScrollLocking={false} aria-label=\\"Pick a date\\" />
  <DatePickerContainer id=\\"searchDateRangeEndDate\\" label=\\"filters.period.custom.end_date\\" okLabel=\\"filters.period.custom.dialog.ok\\" cancelLabel=\\"filters.period.custom.dialog.cancel\\" portal={true} fullWidth={false} icon={{...}} defaultValue={{...}} value={{...}} locales=\\"en-GB\\" minDate={{...}} maxDate={{...}} onChange={[Function: onChange]} animateInline={true} previousIcon={{...}} nextIcon={{...}} autoOk={false} yearsDisplayed={100} defaultCalendarMode=\\"calendar\\" DateTimeFormat={[Function: DateTimeFormat]} okPrimary={true} cancelPrimary={true} closeOnEsc={true} closeYearOnSelect={false} disableScrollLocking={false} aria-label=\\"Pick a date\\" />
</div>"
`;
